enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,362 @@
+                     |    / V \                  ‖           |
+                     ----/| V |\-----------------‖------------
+                          | V |                  ‖
+                   Reverse| V |Tunnel            ‖ Install/Operate
+                          | V |                  ‖ 
+ -------------------------| V |-------------     ‖
+ | Managed-Cluster        | V |            |     ‖
+ |                        | v |            |     ‖
+ | -----------------   ---------------     |     ‖
+ | | KubeApiserver |< <| Proxy Agent | <==========
+ | -----------------   ---------------     |
+ -------------------------------------------
+
+```
+
+- A proxy-agent that establishes a long connection towards the proxy-server
+  and register itself w/ the cluster's identity.
+- A proxy-server that receives handshakes from proxy-agent and maintains 
+  tunnel over the long connection.
+- A proxy-client requesting the target kube-apiserver through the tunnel
+  over either websocket HTTP-CONNECT or grpc protocol.
+- An addon controller automates the installation of proxy servers and proxy 
+  agents by consuming the corresponding configuration custom resource. And
+  optionally the proxy-agents can aggregate and report the status upwards
+  so that admins can checkout their healthiness easily w/o even touching
+  the managed-clusters.
+
+
+#### Components
+
+##### Cluster Proxy Operator
+
+According to the latest design and implementation of [addon-framework](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/8-addon-framework),
+we will bring a new agent controller named `cluster-proxy-operator` that
+manages the lifecycle of the proxy components by watching the configuration
+under a new group of api `proxy.addon.open-cluster-management.io/v1alpha1`:
+
+- __ManagedProxyServer__: the configuration and status of proxy-servers.
+
+- __MangedProxyAgent__: the configuration and status of proxy-agents.
+
+Additionally, the operator will be list-watching the standard `ManagedClusterAddon`
+resource and automatically installs proxy-agents w/ the expected options
+upon discovering the addon in each cluster's namespace.
+
+
+##### ManagedProxyServer
+
+The `ManagedProxyServer` is supposed to be a cluster-scoped resource and holds 
+information for provisioning an active proxy-server instance. An example will 
+be:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedProxyServer
+metadata:
+  name: default # Or any unique name, will be the name of the workload
+spec:
+  replicas: 1
+  certificates:
+    signingSANs: 
+      - "1.1.1.1"
+      - "example.com"
+    rotation:
+      expiringDays: 365
+      forceReloadDays: 200
+  proxy: 
+    mode: [http-connect|grpc|uds] 
+  deploy:
+    namespace: open-cluster-management
+    agentAuthentication:
+      namespace: <validating namespace>
+      serviceAccount: <validating service-account>
```

Comment:
I think it could be a fixed name, will you expect user mutate this field?

Reply:
firstly for clarification, the things under `agentAuthentication` is optional and meant for authorizing the agent handshakes. e.g. proxy-server should only accepts agents mounting expected service-accounts. i dont completely understand the purpose of such additional authorization b/c the agent handshakes are already secured/authenticated by mTLS layer. 

> will you expect user mutate this field?

i picked a few outstanding configuration options from the whole spill. there're more flags such as loggings. i think we can start by keeping it in case any user will need additional authorization between server <-> agent. wdyt?